runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -348,19 +385,58 @@ function LanguageTree:parse()
     end
   end
 
+  return query_time
+end
+
+--- Run a parse on the language tree.
+---
+---@param range boolean|Range|nil: Parse this range in the parsers source.
+---    This only affects regions in the source which have an explicit
+---    range defined which includes: regions set with `set_included_regions()`,
+---    and, injections.
+---    Set to `true` to run a complete parse of the source. (This can be slow.)
+---    Set to `false|nil` to only parse regions with undefined ranges (typically
+---    this is the root tree).
```

Comment:
```suggestion
--- Recursively parse all regions in the language tree using a |treesitter-parser|
--- for the corresponding language and run injection queries on the parsed trees
--- to determine whether child trees should be created and parsed.
---
--- Any region with empty range (`{}`, typically only the root tree) is always parsed; 
--- otherwise (typically injections) only if it intersects {range} (or if {range} is `true`).
---
---@param range boolean|Range|nil: Parse this range in the parser's source.
---    Set to `true` to run a complete parse of the source (Note: Can be slow!)
---    Set to `false|nil` to only parse regions with empty ranges (typically
---    only the root tree without injections).
```

Omitting mentioning `set_included_regions()` as this is `@private` (for now).